File: LeetCode/cpp/217_contains_duplicate.cc
/*
Difficulty: Easy
*/

#include <unordered_set>
#include <vector>

class Solution {
 public:
  bool containsDuplicate(const std::vector<int>& nums) {
    std::unordered_set<int> seen;
    for (const int num : nums) {
      if (seen.find(num) != seen.cend())
        return true;
      else
        seen.insert(num);
    }

    return false;
  }
};

File: LeetCode/rust/217_contains_duplicate.rs
/*
Difficulty: Easy
*/

use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut seen = HashSet::new();

        for num in nums {
            match seen.contains(&num) {
                true => return true,
                false => seen.insert(num),
            };
        }
        return false;
    }
}
